app/models/automation_request.rb

```diff
@@ -79,4 +66,18 @@ def my_role(_action = nil)
   def log_request_success(_requester_id, _mode)
     # currently we do not log successful automation requests
   end
+
+  def self.parse_out_objects(parameters)
+    parameters = parameters.dup
+    object_hash = parameters.select { |key, _v| key.to_s.include?(MiqAeEngine::MiqAeObject::CLASS_SEPARATOR) }
+    parameters.extract!(*object_hash.keys)
+    return object_hash, parameters
+  end
+  private_class_method :parse_out_objects
+
+  def self.build_attrs(parameters, user)
+    object_parameters, parameters = parse_out_objects(parameters)
+    MiqRequestWorkflow.parse_ws_string(parameters).merge!(object_parameters).tap { |attrs| attrs[:userid] = user.userid }
```

Comment:
While the `tap` is nice this line is getting a little long and hard to read.  I would not mind using a multi-line block here.

```ruby
MiqRequestWorkflow.parse_ws_string(parameters).merge!(object_parameters).tap do |attrs| 
  attrs[:userid] = user.userid
end
```

Reply:
phooey